File: models/Ticket.js
const mongoose = require('mongoose');

const TicketSchema = new mongoose.Schema({
  platform: {
    type: String,
    required: true
  },
  category: {
    type: String,
    required: true
  },
  subject: {
    type: String,
    required: true
  },
  description: {
    type: String,
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  status: {
    type: String,
    enum: ['Open', 'Closed', 'Pending'],
    default: 'Open'
  },
  assignedTo: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  },
  messages: [
    {
      user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
      message: { type: String, required: true },
      dateTime: { type: Date, default: Date.now }
    }
  ],
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Ticket', TicketSchema);

File: SwiftIOSApp/Models/Ticket.swift
import Foundation

struct Ticket: Codable, Identifiable {
    let id: String
    let userId: String
    var platform: String?
    var category: String
    var subject: String
    var description: String?
    var status: String?
    var assignedTo: String?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case userId
        case platform
        case category
        case subject
        case description
        case status
        case assignedTo
    }
}
